R/dependencies.R

```diff
@@ -122,15 +122,20 @@ snapshotDependencies <- function(appDir, implicit_dependencies=c()) {
       }
       repository <- package.repo$url
     } else {
-      warning(sprintf("Unable to find repository URL for package %s", pkg),
-              immediate. = TRUE)
+      warning(sprintf("Unable to determine the repository for package %s", pkg),
+              call. = FALSE, immediate. = TRUE)
     }
     data.frame(Source = source, Repository = repository)
   })
   records[, c("Source","Repository")] <- do.call("rbind", tmp)
   return(records)
 }
 
+# Return TRUE when the source indicates that a package comes from a source control system.
+is.scm.source <- function(source) {
```

Comment:
I'm going to be super nitpicky here, just because it's R and weird. `method.class` is used for S3 dispatch... it seems ( from `random.repo.name` below) that there is some convention for this, but would `is_scm_source` work? I had a heck of a time extricating the dot convention out of a package one time after hitting weird S3 issues and I have never forgotten the pain 😅 Not a big deal to me. Just thought I'd mention it 😄